docs/docsite/rst/dev_guide/developing_modules_network.rst

```diff
@@ -0,0 +1,51 @@
+.. _network_dev_guide:
+
+**************************
+Developing network modules
+**************************
+
+.. contents::
+   :local:
+
+.. _developing_modules_network:
+
+Network connection plugins
+==========================
+
+Each network connection plugin has a set of implementation plugins which provide a specific implementation of that connection for a particular set of devices.
+
+Public methods of these plugins may be called on the connection proxy object from the module just as connection methods can.
+
+.. _developing_plugins_httpapi:
+
+Developing HttpApi plugins
+--------------------------
+
+:ref:`HttpApi plugins <httpapi_plugins>` serve as adapters for various HTTP(S) APIs for use with the ``httpapi`` connection plugin. They should implement a minimal set of convenience methods tailored to the API you are attempting to use.
+
+Specifically, there are a few methods that the ``httpapi`` connection plugin expects to exist.
+
+Making requests
+^^^^^^^^^^^^^^^
+
+The ``httpapi`` connection plugin has a ``send()`` method, but an HttpApi plugin needs a ``send_request(self, data, **message_kwargs)`` method as a higher-level wrapper to ``send()``. This method should prepare requests by adding fixed values like common headers or URL root paths, and may do more complex work such as turning data into formatted payloads, or determining which path or method to request. It may then also unpack responses to be more easily consumed by the caller.
```

Comment:
What's the difference between httpapi and HttpApi?

The second sentence could be cleaned up. Not sure if it's a run-on per ce, but it is too wordy.

Also, this hits one of my pet pieves - using the word 'consume'.

Reply:
Yeah, don't capitalize, there's no value IMO, only confusing.